artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -881,13 +919,17 @@ public void addHead(final MessageReference ref, boolean scheduling) {
       enterCritical(CRITICAL_PATH_ADD_HEAD);
       synchronized (this) {
          try {
-            if (!scheduling && scheduledDeliveryHandler.checkAndSchedule(ref, false)) {
-               return;
-            }
+            enforceRing(ref, scheduling);
```

Comment:
can a if check be wrapped around this, to avoid the method (which method is large and unlikly to be inlined) call when ringsize is disabled (-1)

Reply:
(nit)

Reply:
Will do.